src/poetry/packages/locker.py

```diff
@@ -128,9 +128,21 @@ def locked_repository(self) -> Repository:
                 # Old lock so we create dummy files from the hashes
                 hashes = cast("dict[str, Any]", metadata["hashes"])
                 package.files = [{"name": h, "hash": h} for h in hashes[name]]
+            elif source_type in {"git", "directory", "url"}:
+                package.files = []
             else:
                 files = metadata["files"][name]
-                package.files = files
+                if source_type == "file":
+                    if len(files) > 1:
```

Comment:
sorry I wasn't clear earlier - suggest remove this check too.  At best it saves us iterating through a list of length 0 or 1, at worst it causes us to put a bad file here and later refuse to install the package because of hash mismatch.